src/components/patch-assignment-form/components/form-buttons.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { Dispatch, SetStateAction } from "react";
+
+import { Patch } from "@mtfh/common/lib/api/patch/v1";
+
+export const CancelReassignmentButton = ({
+  onClick,
+}: {
+  onClick: Function;
+}): JSX.Element => {
+  return (
+    <button
+      data-testid="cancel-reassignment-button"
+      className="govuk-button lbh-button"
+      style={{ marginTop: 0 }}
+      onClick={(e) => {
+        e.preventDefault();
+        onClick();
+      }}
+    >
+      Cancel
+    </button>
+  );
+};
+
+export const ReassignButton = ({ onClick }: { onClick: Function }): JSX.Element => {
+  return (
+    <button
+      data-testid="reassign-button"
+      className="govuk-button lbh-button"
+      style={{ marginTop: 0 }}
+      onClick={(e) => {
+        e.preventDefault();
+        onClick();
+      }}
+    >
+      Reassign
+    </button>
+  );
+};
+
+interface AssignButtonProps {
+  setReassigningPatch: Dispatch<SetStateAction<Patch | null>>;
+  reassigningThisEntity: boolean;
+  officerName: string;
+  onClick: Function;
+}
+export const AssignButton = ({
+  setReassigningPatch,
+  reassigningThisEntity,
+  officerName,
+  onClick,
+}: AssignButtonProps) => {
+  if (reassigningThisEntity) {
+    return <CancelReassignmentButton onClick={() => setReassigningPatch(null)} />;
+  }
+  return (
+    <button
+      data-testid="assign-button"
+      className="govuk-button lbh-button"
+      style={{ marginTop: 0, maxHeight: "2.5em" }}
+      onClick={(e) => {
+        e.preventDefault();
+        onClick();
+      }}
+    >
+      Assign {officerName}
+    </button>
+  );
+};
```

Comment:
My main objection to this is the name lies. This is not the `AssignButton`, but instad the `AssignButPossiblyReassignButton`. 

When you've used this button, you're already conditionally checking `reassigningPatch`. Why not check for `reassigningThisEntity`?

Reply:
Good point, split the buttons in 2284401d08c5757f96b0a1ecc3ddfc18d68015a8